### DISCUSSION TITLE
skipMissingProperties issue?

### DISCUSSION THREAD

**Discussion by chkp-erang** [typestack/class-validator#1907] [Questions & Answers] [answered]
Hi,
I am using GQL server and use class-validator to perform validation on the query/mutation input.

It appears that when I upgrade from version 0.13.2 to 0.14.0, I am getting the following error:
{
  "unknownValue": "an unknown value was passed to the validate function"
}
The error is given on non mandatory leaf field defined on inner object (level 4).
When debugging the code, it looks like it use skipMissingProperties option.
Can someone assist?

**Comment by braaar** [comment, marked as answer]
`forbidUnknownValues ` defaults to `true` in 0.14.0 onwards. If this breaks your validation, please consider refactoring your code so that this problem doesn't occur. You are possibly risking some unknown data sneaking into your program undetected, so to speak.

**Reply by braaar** [reply]
If that doesn't fix it for you, see #1873

**Comment by chkp-erang** [comment]
Hi Brage,
Thank you for the reply. 
I adjust my code to make class validator be aware the classes that it failed on and the optional fields. so now everything is working well with forbidUnknownValues = true

Note that I didn't understood the meaning of this flag from the documentation, maybe worth explaining it as now this is the default.
